cpp/src/parquet/stream_reader_test.cc

```diff
@@ -261,7 +275,176 @@ TEST_F(TestStreamReader, ValueChecking) {
     ASSERT_FLOAT_EQ(f, TestData::GetFloat(i));
     ASSERT_DOUBLE_EQ(d, TestData::GetDouble(i));
   }
+  ASSERT_EQ(reader_.current_row(), TestData::num_rows);
+  ASSERT_EQ(i, TestData::num_rows);
+}
+
+TEST_F(TestStreamReader, SkipRows) {
+  bool b;
+  std::string s;
+  std::array<char, 4> char_array;
+  char c;
+  int8_t int8;
+  uint16_t uint16;
+  int32_t int32;
+  uint64_t uint64;
+  std::chrono::microseconds ts_us;
+  float f;
+  double d;
+  std::string str;
+
+  const int num_rows_to_read = 3;
+  const int num_rows_to_skip = 13;
+  int num_rows_read = 0;
+  int i = 0;
+
+  // Skipping zero rows is ok.
+  //
+  ASSERT_EQ(0, reader_.SkipRows(0));
+
+  ASSERT_EQ(false, reader_.eof());
+  ASSERT_EQ(0, reader_.current_row());
+  ASSERT_EQ(TestData::num_rows, reader_.num_rows());
+
+  while (!reader_.eof()) {
+    for (int j = 0; !reader_.eof() && (j < num_rows_to_read); ++i, ++j) {
+      reader_ >> b;
+      reader_ >> s;
+      reader_ >> c;
+      reader_ >> char_array;
+      reader_ >> int8;
+      reader_ >> uint16;
+
+      // Not allowed to skip row once reading columns has started.
+      ASSERT_THROW(reader_.SkipRows(1), ParquetException);
+
+      reader_ >> int32;
+      reader_ >> uint64;
+      reader_ >> ts_us;
+      reader_ >> f;
+      reader_ >> d;
+      reader_ >> EndRow;
+      num_rows_read += 1;
+
+      ASSERT_EQ(b, TestData::GetBool(i));
+      ASSERT_EQ(s, TestData::GetString(i));
+      ASSERT_EQ(c, TestData::GetChar(i));
+      ASSERT_EQ(char_array, TestData::GetCharArray(i));
+      ASSERT_EQ(int8, TestData::GetInt8(i));
+      ASSERT_EQ(uint16, TestData::GetUInt16(i));
+      ASSERT_EQ(int32, TestData::GetInt32(i));
+      ASSERT_EQ(uint64, TestData::GetUInt64(i));
+      ASSERT_EQ(ts_us, TestData::GetChronoMicroseconds(i));
+      ASSERT_FLOAT_EQ(f, TestData::GetFloat(i));
+      ASSERT_DOUBLE_EQ(d, TestData::GetDouble(i));
+    }
+    ASSERT_EQ(num_rows_to_skip, reader_.SkipRows(num_rows_to_skip));
```

Comment:
sorry if I missed it but do you have a test case for a skip that hits EOF with a smaller number of rows?

Reply:
Yes the SkipRows test hits EOF after skipping the last 13 rows.
This is because there are 2000 rows and the test reads 3 rows and the skips the next 13 rows repeatedly.
Since 2000 / 16 is exactly 125 the last 13 rows skipped trigger an EOF.